### PR TITLE
PYTHON-3338 Add versionadded to docs for key management APIs

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: pymongo/encryption.py [hunk 1]
```diff
@@ -260,15 +260,20 @@ def close(self):
...
 
 class RewrapManyDataKeyResult(object):
+    """Result object returned by a ``rewrap_many_data_key`` operation.
+
+    .. versionadded:: 4.2
+    """
+
     def __init__(self, bulk_write_result: Optional[BulkWriteResult] = None) -> None:
-        """Result object returned by a ``rewrap_many_data_key`` operation.
+        """Create a ``RewrapManyDataKeyResult`` for a ``rewrap_many_data_key`` operation.
```
Comment (ShaneHarvey) [inline pymongo/encryption.py:269]: Following the pattern in the other result classes (eg UpdateResult) we can remove the `__init__` docstring. Users aren't expected to create this class anyway.

File: pymongo/encryption.py [hunk 2]
```diff
@@ -260,15 +260,20 @@ def close(self):
 
 
 class RewrapManyDataKeyResult(object):
+    """Result object returned by a ``rewrap_many_data_key`` operation.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pymongo/encryption.py:263]: Done

File: pymongo/encryption.py (hunk 1 above)
Comment (blink1073) [reply pymongo/encryption.py:269]: Done

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/encryption.py [hunk 3]
```diff
@@ -260,16 +260,12 @@ def close(self):
 
 
 class RewrapManyDataKeyResult(object):
-    def __init__(self, bulk_write_result: Optional[BulkWriteResult] = None) -> None:
-        """Result object returned by a ``rewrap_many_data_key`` operation.
+    """Result object returned by a :meth:`rewrap_many_data_key` operation.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pymongo/encryption.py]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**